### ISSUE TITLE
Changing Movable of node does not work

### ISSUE THREAD

**Issue by Epictek** [excubo-ag/Blazor.Diagrams#57]
I have a list of nodes rendered using a foreach, I have a button which should flip a bool which is used by the the nodes to toggle movable, toggle doesn't seem to have an effect until I zoom in . I think it may have something to do with the default=true in NodeBase for movable?

I can upload a demo of the issue later if required.

**Comment by Epictek** [comment]
```C#
@page "/performance"
<SourceCodeLink Repository="Blazor.Diagrams" Page="@(nameof(Performance))" />

<Diagram>
    <Nodes>
        @foreach (var state in states)
        {
            <!-- Builtin node, node-type (i.e. shape) specified by Type property -->
            <Node @key="state.Id" Id="@state.Id" @bind-X="state.X" @bind-Y="state.Y" Type="NodeType.Rectangle" Movable="@Movable">
                State @state.Id
            </Node>
        }
    </Nodes>
    <Links>
        @foreach (var link in links)
        {
<Link @key="link.Source.NodeId" Source="@link.Source" Target="@link.Target" />}
    </Links>
</Diagram>

<button @onclick="() => { Movable =!Movable; StateHasChanged(); }">@Movable.ToString()</button>

@code { 
    bool Movable = true;
    private const int n = 10;
    public class MyNodeDefinition
    {
        public string Id { get; set; }
        public double X { get; set; }
        public double Y { get; set; }
    }
    public class MyLinkDefinition
    {
        public NodeAnchor Source { get; set; }
        public NodeAnchor Target { get; set; }
        public LinkType Type { get; set; }
    }
    List<MyNodeDefinition> states = Enumerable.Range(0, n * n).Select(i => new MyNodeDefinition
    {
        Id = $"state{i}",
        X = 150 * (i % n),
        Y = 150 * (i / n)
    }).ToList();
    List<MyLinkDefinition> links = Enumerable.Range(0, n * n / 4).Select(i => new MyLinkDefinition
    {
        Source = new NodeAnchor { NodeId = $"state{2 * i}", Port = Position.East },
        Target = new NodeAnchor { NodeId = $"state{2 * i + 1}", Port = Position.West }
    }).Concat(Enumerable.Range(0, n * n / 4).Select(i => new MyLinkDefinition
    {
        Source = new NodeAnchor { NodeId = $"state{2 * n * n / 4 + 2 * i}", Port = Position.NorthEast },
        Target = new NodeAnchor { NodeId = $"state{2 * n * n / 4 + 2 * i + 1}", Port = Position.SouthWest }
    })).ToList(); }
```

**Comment by stefanloerwald** [comment]
Hi @Epictek,

this issue is quite a tricky one. I can reproduce the issue, but haven't been able to find the root cause yet. I will continue to investigate.

Stefan

**Comment by stefanloerwald** [comment]
The issue is now identified and resolved. It was tricky to find, as it actually had nothing to do with `Movable`. A flag that was necessary in a previous version to prevent costly re-rendering of parts of the components had now prevented the rendering that would toggle the `Movable` state. Fixed in [version 3.0.2](https://www.nuget.org/packages/Excubo.Blazor.Diagrams/3.0.2) by commit  a72b798.
